tooling/metamodel-generator/src/main/java/org/hibernate/processor/HibernateProcessor.java

```diff
@@ -373,7 +382,7 @@ private void processClasses(RoundEnvironment roundEnvironment) {
 			try {
 				final AnnotationMetaEntity metaEntity =
 						AnnotationMetaEntity.create( typeElement, context,
-								parentMetadata( typeElement, context::getMetaEntity ) );
+								parentMetadata( typeElement, context::getMetaEntity ), null );
```

Comment:
How about overloading `AnnotationMetaEntity.create()` instead of passing `null`?